### DISCUSSION TITLE
Snappy user change password

### DISCUSSION THREAD

**Discussion by Fraleu** [the-djmaze/snappymail#729] [Q&A]
I'm looking on addon to allow user to change his mailbox password alone.
could you help me to show me how to use as addon ?*
Thanks for your help

**Comment by MI-KY** [comment]
There exist multiple plugins for this. Could you tell us the exact name of the plugin you're trying to use / install?
Please tell also where you're stuck so maybe someone can help you more specifically.

**Reply by Fraleu** [reply]
Thankls MI-KY
I'm looking for plugin suggestion to allow user change password using SnappyMail
I have no existing pluins target for now. What should be your suggestion today to allow user to change password on snappymail services ?
Let me know
Best regards

**Comment by MI-KY** [comment]
Personally I don't use such a functionality but you could try the "change-password" extension (https://github.com/the-djmaze/snappymail/tree/master/plugins/change-password) available directly in the Admin Panel -> Extensions.
The best thing would be to try out this extension and maybe report your experiences here for others 👍 🙂

**Comment by Fraleu** [comment]
yes i will try and report experiences.
but now how i can install or find how to do or instruction manual of installation ?

**Reply by MI-KY** [reply]
Actually the documentation is work in progress, but a bit it is already described here: https://github.com/the-djmaze/snappymail/wiki/Developer-Documentation#general-information

You have to enter into the admin panel, choose "extensions", activate the plugin management and then install the needed plugin. When installed you see a gear symbol beside the name and there you can configure the plugin...

**Reply by Fraleu** [reply]
Sorry MI-KY i'don't have "extensions" on my admin panel, is it normal ?
see below cyberpanel version management and my menu screenshot.

![cyberpanel#1](https://user-images.githubusercontent.com/119486565/205041806-ffe162cb-3e46-4308-a960-47ae0024e104.png)
![cyberpanel#2](https://user-images.githubusercontent.com/119486565/205041840-08d422b9-2970-43a9-8c3d-40b836b55707.png)

**Comment by Fraleu** [comment]
and nothing to install plugin on Plugins/Installed page 
![cyberpanel#3](https://user-images.githubusercontent.com/119486565/205042305-50a4e7ef-2dfb-4f47-9ad8-fed350b6edc1.png)

**Reply by MI-KY** [reply]
Hi,
this is not the Admin Panel of SnappyMail but seems to be the Admin Panel of your VPS / Cloud service...
How did you install and configure SnappyMail?

**Comment by Fraleu** [comment]
oh sorry you right that's Cyberpanel interface.
I install SnappyMail with Standrd Cyberpanel installation
so i don't know how to get access to SnappyMail panel interface ?

**Reply by MI-KY** [reply]
Unfortunately I don't know anything about Cyberpanel... But have a look at https://github.com/the-djmaze/snappymail/wiki/Installation-instructions#general-instructions maybe the step you need to do is to add ?admin to the url... But this seems to be a bit specific to Cyberpanel.

**Comment by Fraleu** [comment]
How to generate new admin password by visiting https://example.com:8090/snappymail/?admin
any idea  on following interface ?
![snappymail#1](https://user-images.githubusercontent.com/119486565/205121849-d9403157-51d9-4450-9e34-443c1aa97527.png)

**Reply by MI-KY** [reply]
https://github.com/the-djmaze/snappymail/wiki/Installation-instructions#general-instructions
and
https://github.com/the-djmaze/snappymail/wiki/Installation-instructions#admin-password

**Comment by Fraleu** [comment]
Great
Thanks MI-KY 
with your link i finally get Admin SnappyMail Panel
So installing Change Password extension easilly.
I got Change password on user snappy configuration parameters 
BUT
Extension not able to save password change ....
I don't change anything on Extension parameters
![snappymail#2](https://user-images.githubusercontent.com/119486565/205140894-cd691772-ae6b-44e1-980b-5c48cfc1489c.png)

**Comment by Fraleu** [comment]
I certainlly miss something ... any idea ?
![snappymail#3](https://user-images.githubusercontent.com/119486565/205141457-9ed56738-7ab4-4eea-9be7-50c0be5b02d4.png)

**Reply by MI-KY** [reply]
Glad you did it 👍.
As said before I don't have experience with this plugin... So I can't help you... Maybe have a look into the log files, but as said, no idea...

**Comment by Fraleu** [comment]
Any body else can help with "Change password" extension for snapyMail ?
Still not working today.
![image](https://user-images.githubusercontent.com/119486565/210089525-b2ca6a0c-c630-4d0d-b579-039606b697e8.png)

**Reply by the-djmaze** [reply]
You need to know where and how Cyberpanel stores your IMAP and SMTP password.
From there on we can tell how to setup the change-password plugin

**Comment by Fraleu** [comment]
Hello djmaze,
I just got reply from my contact in Cyberpanel
==> They are stored in mysql database 
let me know on follow
BR


Tuesday, January 3, 2023, 3:49:54 PM, you wrote:

> You need to know where and how Cyberpanel stores your IMAP and SMTP password.
> From there on we can tell how to setup the change-password plugin
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you authored the thread.

-- 
Best regards,
Franck sur Miramont.bzh

**Reply by the-djmaze** [reply]
https://community.cyberpanel.net/t/tutorial-how-to-allow-your-snappymail-users-to-change-their-mailbox-passwords/38084
